Release 1.56 Blog Post (#3547)
FusionAuth/fusionauth-site: astro/src/content/blog/announcing-fusionauth-1-56.mdx

---
publish_date: 2025-03-04
title: Announcing FusionAuth Version 1.56 - The SSO Snake
description: FusionAuth v1.55.1 "The Holiday Hippo" brings major updates including SAML encryption support, enhanced API key security with via expiration, configurable lambda timeouts, and more. 
authors: Brad McCarty
image: /img/blogs/release-1-56/fusionauth-1-56.png
categories: Product
tags: SSO, Custom Domains, API Key Security
excerpt_separator: "{/* more */}"
---

> 

Reviewer: ```suggestion
import Aside from '/src/components/Aside.astro';

```

This is used below.

> We recently released FusionAuth version 1.56.0. This version focuses on API security, and it introduces some new flexibility in how you can use SSO sessions within FusionAuth.

Reviewer: ```suggestion
We recently released FusionAuth version 1.56.0. This version focuses on API security, and it introduces some new flexibility in how you can use SSO sessions within FusionAuth. We also have custom domains for every hosting plan.
```

{/* more */}

Our fearless Senior Director of Developer Relations ([welcome to the team, Kim](https://www.linkedin.com/posts/fusionauth_this-week-we-are-super-excited-to-welcome-activity-7301000700145131520-FkVA?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAGscwkBMMEEWM3JSitpba873aUwI_0jz6Q)!) has dubbed this release “The SSO Snek.” Our slithering friend is here to sneak past expired SSO sessions and ensure a seamless experience for your users.

![The SSO Snek is a sneaky little guy](/img/blogs/release-1-56/sso-snake.jpg)

And now, on to the good stuff!

## API Key Security

> In version 1.56.0, we’ve improved API key security by introducing hashed storage. This builds on our [1.55.1 release](/blog/announcing-fusionauth-1-55) that added expiration times. With 1.56.0, newly-created API keys are now optionally secured before being stored, reducing risks in case of a database compromise. We want to stress that the default behavior is not changing, so your existing keys will continue to work. If you want your keys secured, you will need to create new ones.

Reviewer: I can't see the comments Andy made about hashing vs securing, so feel free to ignore if you've already considered these things.

> In version 1.56.0, we’ve improved API key security by introducing hashed storage. This builds on our [1.55.1 release](/blog/announcing-fusionauth-1-55) that added expiration times. With 1.56.0, newly-created API keys are now optionally secured before being stored, reducing risks in case of a database compromise. We want to stress that the default behavior is not changing, so your existing keys will continue to work. If you want your keys secured, you will need to create new ones.

Author: You're right. The comments in the doc discussed this. Resolving this one without a commit.

A note of caution: If you secure the API key, retrieving its value after creation is no longer possible due to this security measure. So if you need the key to be accessible in plaintext, make sure to copy it on creation or choose not to secure it.

## Custom Domains for Everyone (on FusionAuth Cloud)

Custom domains allow you to keep your domain name front and center during the authentication process. So instead of a customer seeing piedpiper.fusionauth.io they could see auth.piedpiper.com. We’ve supported custom domains since 2021 (and unlimited domains [since 2023](/blog/introducing-unlimited-custom-domains-for-fusionauth-cloud)), but they were only available to Business hosting and above. As of 1.56.0, all new FusionAuth Cloud deployments on our next-gen platform have custom domains enabled:

- Basic Hosting: 1 Custom Domain
- Business Hosting: 5 Custom Domains

> - HA Hosting: 20 Custom Domains

Reviewer: ```suggestion
- HA Hosting: 20 Custom Domains (unlimited if you have an Enterprise plan)
```

>   - (Unlimited Custom Domains supported with an Enterprise plan)

Reviewer: ```suggestion
```

Reviewer: This formatted funny so I collapsed them.

> Note: Existing customers on our legacy cloud platform will not have access to the new custom domain limits. If you would like to enable custom domains, please contact support to help you migrate to our newest implementation of FusionAuth Cloud.

Reviewer: ```suggestion
<Aside type="note">
Existing customers may not have access to the new custom domains, and you need to be on a more recent version of FusionAuth. If you would like to use custom domains, please [contact support](https://account.fusionauth.io/account/support/) to learn more.
</Aside>
```

Reviewer: In general we've been opaque about the differences in clouds, since customers shouldn't care.

authors: Brad McCarty
image: /img/blogs/release-1-56/fusionauth-1-56.png
categories: Product
tags: SSO, Custom Domains, API Key Security
excerpt_separator: "{/* more */}"
---
import Aside from '/src/components/Aside.astro';

We recently released FusionAuth version 1.56.0. This version focuses on API security, and it introduces some new flexibility in how you can use SSO sessions within FusionAuth. We also have custom domains for every hosting plan.

{/* more */}

Our fearless Senior Director of Developer Relations ([welcome to the team, Kim](https://www.linkedin.com/posts/fusionauth_this-week-we-are-super-excited-to-welcome-activity-7301000700145131520-FkVA?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAGscwkBMMEEWM3JSitpba873aUwI_0jz6Q)!) has dubbed this release “The SSO Snek.” Our slithering friend is here to sneak past non existent SSO sessions and ensure a seamless experience for your users.

![The SSO Snek is a sneaky little guy](/img/blogs/release-1-56/sso-snake.jpg)

And now, on to the good stuff!

## API Key Security

> In version 1.56.0, we’ve improved API key security by introducing hashed storage. This builds on our [1.55.1 release](/blog/announcing-fusionauth-1-55) that added expiration times. With 1.56.0, newly-created API keys are now optionally secured before being stored, reducing risks in case of a database compromise. We want to stress that the default behavior is not changing, so your existing keys will continue to work. If you want your keys secured, you will need to create new ones.

Reviewer: ```suggestion
In version 1.56.0, we’ve improved API key security by introducing secured storage. This builds on our [1.55.1 release](/blog/announcing-fusionauth-1-55) that added expiration times. With 1.56.0, newly-created API keys are now optionally secured before being stored, reducing risks in case of a database compromise. We want to stress that the default behavior is not changing, so your existing keys will continue to work. If you want your keys secured, you will need to create new ones.
```
